### PR TITLE
Update Debian ISO to Debian-12.0.0

### PR DESCRIPTION
**Pull request by alexpeterson91**

### CODE REVIEW

**Comment by 3alpha** [comment]
🚀

File: iso/preseeds/preseed.cfg
Comment (3alpha) [inline iso/preseeds/preseed.cfg:28]: Something to be aware of regarding non-free: https://www.debian.org/releases/bookworm/amd64/release-notes/ch-information.en.html#non-free-split

We need to figure out which exactly repo do we need. It should be only firmware, but I am not completly sure.

File: iso/preseeds/preseed_unattended.cfg
Comment (3alpha) [inline iso/preseeds/preseed_unattended.cfg:65]: Same comment as before.

File: .github/workflows/artifacts.yml
Comment (3alpha) [inline .github/workflows/artifacts.yml:6]: I am also often guily of this, but we should keep our PRs as single issue as we can. Preferably, this should be in a separate PR which we can merge without going through all the testing necessary for the other stuff. 

Keep it as it is now, this comment is just something to consider for the future.

File: iso/preseeds/preseed.cfg
Comment (alexpeterson91) [reply iso/preseeds/preseed.cfg:28]: I think they changed the `non-free` firmware to a new module called `non-free-firmware` 
Explanation of the change of  the `non-free` to `non-free-firmware` : https://unix.stackexchange.com/questions/736065/how-do-i-install-non-free-firmware-in-debian-12-bookworm
official example: https://www.debian.org/releases/bookworm/example-preseed.txt
someone's filled out example with apt packages listed etc: https://preseed.debian.net/debian-preseed/bookworm/amd64-main-full.txt

File: iso/preseeds/preseed_unattended.cfg
Comment (alexpeterson91) [reply iso/preseeds/preseed_unattended.cfg:65]: same comment as above for me too. I will continue to look stuff over make tweaks and test again with my node.  
If anyone else in @dappnode/core has suggestions please  let me know and i can try it out!.

File: .github/workflows/artifacts.yml
Comment (alexpeterson91) [reply .github/workflows/artifacts.yml:6]: yeah i just did it because i had to update the workflows to make it create ISOs with Bookworm in the name not bullseye, so i just slipped it in because the typos had been driving me nuts.  I try to keep things separate but just had to do something since i already had to edit the workflows for this branch anyway., but yeah i do it too often mixing in extra changes in PRs, ill try and keep to proper etiquette in my PRs and commits

**Review by pablomendezroyo — DISMISSED** [review]

File: iso/preseeds/preseed.cfg
Comment (3alpha) [reply iso/preseeds/preseed.cfg:28]: As far as I understood it, it wasn't renamed but what was before `non-free` now has been split in `non-free` and `non-free-firmware`.

In our case, I think we just use firmware stuff.

Comment (alexpeterson91) [reply iso/preseeds/preseed.cfg:28]: ah ok i read it a different way but i think you're correct, gonna try some new settings and try a reinstall on my test node see if it works.  if you have any specific suggestions for changes to the preseed configs please go ahead and make/review said changes and i can test it too, this stuff i literally tried to learn in like an hour a couple days ago when this was released, so commits are appreciated, @pablomendezroyo approved this PR but it's definitely not ready to merge just yet since we haven't resolved the preseed issues yet. im gonna get to testing some stuff now, but ill likely need help making sure any changes arent gonna break something, but right now im just trying to make sure the package manager works, which hopefully should just need changes to make it not look for CD's i think that issue is because the new bookworm ISOs are net installers but not called or named netinst anymore so the entry `d-i apt-setup/disable-cdrom-entries boolean true` needs to be there as @3alpha mentioned on discord.  Im just not sure if other things need to be added or removed, but i guess well see, cant do much without trying lol.

Comment (alexpeterson91) [reply iso/preseeds/preseed.cfg:28]: so @3alpha i. messed around a bit and got stuck in the installer during select and install software section, couldnt get much from logs, it didnt fail during the preseed either so idk what was up. what do i need to add to keep apt from looking for cd sources and for this line should i remove `non-free` part in this line and the corresponding other preseed and add `non-free-firmware` after  `contrib` instead?

File: iso/preseeds/preseed_unattended.cfg
Comment (alexpeterson91) [inline iso/preseeds/preseed_unattended.cfg:68]:
```suggestion
d-i apt-setup/cdrom/set-failed boolean false
d-i apt-setup/disable-cdrom-entries boolean true
```

File: iso/preseeds/preseed_unattended.cfg
Comment (alexpeterson91) [inline iso/preseeds/preseed_unattended.cfg:65]:
```suggestion
d-i apt-setup/local0/repository string http://deb.debian.org/debian/ bookworm main contrib non-free-firmware
```

File: iso/preseeds/preseed.cfg
Comment (alexpeterson91) [inline iso/preseeds/preseed.cfg:31]:
```suggestion
d-i apt-setup/cdrom/set-failed boolean false
d-i apt-setup/disable-cdrom-entries boolean true
```

File: iso/preseeds/preseed.cfg
Comment (alexpeterson91) [reply iso/preseeds/preseed.cfg:28]:
```suggestion
d-i apt-setup/local0/repository string http://deb.debian.org/debian/ bookworm main contrib non-free-firmware
```

Comment (alexpeterson91) [reply iso/preseeds/preseed.cfg:28]: @3alpha let me know if this looks good, it works so far for me, no issues yet.

**Comment by 3alpha** [comment]
I think we are good, I will just try it once more in my VM before approving.

File: iso/preseeds/preseed.cfg
Comment (alexpeterson91) [reply iso/preseeds/preseed.cfg:28]: @3alpha I think I resolved this conversation before committing the fix in one of my above comments that is used identically  in the `preseed_unattended.cfg` so I assume it supposed to be the same just seem to have closed it as resolved first, committing it now just double check and make sure I'm not missing something other than forgetting to commit it yesterday and instead marking this as closed.  i was checking the diff and thats how I noticed my mistake, just let me know that it was actually my mistake not me ignoring a review, webUI of GH is acting up for me today and I just want you to be sure sure before signing off on this PR.
Totally separate question, what's your hypervisor of choice?, and so you know if any work better than others specifically with Dappnode? im trying to make a final decision before putting my new hardware to work.

**Comment by alexpeterson91** [comment]
@pablomendezroyo @3alpha rerequesting review just because of a rebase, and i want to know for sure if its approved.  But also need to know if this is like ready to merge too or if we need to keep testing it, i dont want to rush it but its just well pretty time critical for some people, and increasingly so.  Let me know! Thanks guys!

**Review by 3alpha — APPROVED** [review]

File: iso/scripts/generate_dappnode_iso_debian.sh
Comment (dsimog01) [inline iso/scripts/generate_dappnode_iso_debian.sh:7]: Good to have this here:
image

Reference: https://cdimage.debian.org/cdimage/unofficial/non-free/cd-including-firmware/

**Comment by dsimog01** [comment]
LGTM!

**Pull request merged**
